src/main/java/htsjdk/samtools/cram/build/ContainerParser.java

```diff
@@ -119,13 +119,14 @@ private static void addSpan(final int seqId, final int start, final int span, fi
 
         final ArrayList<CramCompressionRecord> records = new ArrayList<>(slice.nofRecords);
 
-        int prevStart = slice.alignmentStart;
+        int prevAlignmentStart = slice.alignmentStart;
         for (int i = 0; i < slice.nofRecords; i++) {
             final CramCompressionRecord record = new CramCompressionRecord();
             record.sliceIndex = slice.index;
             record.index = i;
 
-            reader.read(record);
+            reader.read(record, prevAlignmentStart);
```

Comment:
The reader knows when it's in `APDelta` mode and can calculate its `alignmentStart` correctly in either mode because it knows the previous record's start.